plugins/inputs/windows_event_log/wineventlog/wineventlog_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build windows
+// +build windows
+
+package wineventlog
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+var (
+	// common inputs for creating an EventLog.
+	NAME = "Application"
+	// 2 is ERROR
+	LEVELS          = []string{"2"}
+	GROUP_NAME      = "fake"
+	STREAM_NAME     = "fake"
+	RENDER_FMT      = FormatPlainText
+	DEST            = "fake"
+	STATE_FILE_PATH = "fake"
+	BATCH_SIZE      = 99
+	RETENTION       = 42
+)
+
+// TestNewEventLog verifies constructor's default values.
+func TestNewEventLog(t *testing.T) {
+	elog := NewEventLog(NAME, LEVELS, GROUP_NAME, STREAM_NAME, RENDER_FMT, DEST,
+		STATE_FILE_PATH, BATCH_SIZE, RETENTION)
+	assert.Equal(t, NAME, elog.name)
+	assert.Equal(t, uint64(0), elog.eventOffset)
+	assert.Zero(t, elog.eventHandle)
+}
+
+// TestOpen verifies Open() succeeds with valid inputs.
+// And fails with invalid inputs.
+func TestOpen(t *testing.T) {
+	// happy
+	elog := NewEventLog(NAME, LEVELS, GROUP_NAME, STREAM_NAME, RENDER_FMT, DEST,
+		STATE_FILE_PATH, BATCH_SIZE, RETENTION)
+	assert.NoError(t, elog.Open())
+	assert.NotNil(t, elog.eventHandle)
+	assert.NoError(t, elog.Close())
+	// bad name
+	elog = NewEventLog("FakeBadElogName", LEVELS, GROUP_NAME, STREAM_NAME,
+		RENDER_FMT, DEST, STATE_FILE_PATH, BATCH_SIZE, RETENTION)
+	assert.Error(t, elog.Open())
+	assert.Zero(t, elog.eventHandle)
```

Comment:
Question - above, we assert in the happy case that `elog.eventHandle` is not nil, but here we assert that it is zero. For parity, should this be a nil check instead of a zero check?

Reply:
We should use `assert.Zero()` and `assert.NotZero()` because the type is:
```
type EvtHandle uintptr
```

I tried using `assert.Nil()` and `assert.NotNil()`, but got:
```
    wineventlog_test.go:55:
                Error Trace:    wineventlog_test.go:55
                Error:          Expected nil, but got: 0x0
                Test:           TestOpen
--- FAIL: TestOpen (0.03s)
```